lib/src/sdk/dart.dart

```diff
@@ -50,5 +75,17 @@ class DartSdk extends Sdk {
   String get rootDirectory => _rootDirectory;
 
   @override
-  String? packagePath(String name) => null;
+  String? packagePath(String name) {
+    if (!isAvailable) return null;
+    var sdkPackages = _sdkPackages;
+    if (sdkPackages == null) return null;
+
+    var package =
+        sdkPackages.packages.firstWhereOrNull((pkg) => pkg.name == name);
```

Comment:
How about building a map inside the `SdkPackages.fromMap`, so `SdkPackages` has a `SdkPackage? operator[](String packageName)` lookup.
(It's probably a short list, but linear lookup always makes be worry.)